### DISCUSSION TITLE
How the npy file in tutorial 1 come from?

### DISCUSSION THREAD

**Discussion by jacktang** [mathLab/EZyRB#239] [Q&A]
Hello there,

I am learning ROM by following the tutorials and I am puzzled with the npy files. I read the coordination info from `coord.npy` and found it completely disorder. I also searched the discussions and it might come from using Smithers (but I didn't find any example in documentation). And my questions is how to generate the coordination and snapshots npy file? If now I only have coordination csv data and snapshot point data, is it possible to convert to npy file? Thanks a lot!

**Comment by ndem0** [comment]
hi @jacktang, yes you're right, the dataset of the first tutorial comes from Smithers and it is originally created by running finite-element simulations with FEniCS.
But you can use of course CSV files as well, just load them into numpy arrays and you're done!

I do not really get what order are you expecting in the nodes coordinates array, but let us know if you still have doubts!

**Reply by jacktang** [reply]
Hello @ndem0 , thanks for the reply!
I supposed in snapshots based rom one snapshot was organized in some coordinates order. But it is not, right?

**Reply by jacktang** [reply]
@ndem0 I still have question on the parameters' distribution of the snapshots. Take the NavierStokerDataset as example, I plot the values of mu, and it is not the uniform distribution. Could you tell me the paramters distribution? Thanks a lot!

**Reply by mtezzele** [reply]
Probably `np.random.uniform` was used.

**Reply by ndem0** [reply]
Yes, the distribution is just uniform. Why do you write that it is not?

Please, try to put such a question in the proper repository: it would be definitely more helpful for other users/developers to have this conversation on the Smithers page, instead of here.
